src/drivers/imx/ipc.c

```diff
@@ -155,11 +157,30 @@ void ipc_platform_send_msg(struct ipc *ipc)
 	spin_unlock_irq(ipc->lock, flags);
 }
 
+#if CONFIG_HOST_PTABLE
+struct ipc_data_host_buffer *ipc_platform_get_host_buffer(struct ipc *ipc)
+{
+	struct ipc_data *iipc = ipc_get_drvdata(ipc);
+
+	return &iipc->dh_buffer;
+}
+#endif
+
 int platform_ipc_init(struct ipc *ipc)
 {
-	_ipc = ipc;
+#if CONFIG_HOST_PTABLE
```

Comment:
The HOST_PTABLE thing is really just because of the issue mentioned on the previous comment. Otherwise I think we could have managed without (due to linear mapping).